### PR TITLE
Wip/ji/tab icons

### PR DESCRIPTION
**Pull request by jacek-izykowski**
DockWidgets have icons (on tabs and title bars) indicating dock's type.

### CODE REVIEW

File: include/ui/veles_mainwindow.h [hunk 1]
```diff
@@ -37,6 +41,20 @@ namespace ui {
...
+/*****************************************************************************/
+
+class QProxyStyleForDockWidgetWithIconOnTitleBar : public QProxyStyle {
+  Q_OBJECT
+
+public:
+  QProxyStyleForDockWidgetWithIconOnTitleBar(QStyle* default_style);
+
+  void drawControl(QStyle::ControlElement element, const QStyleOption *option,
+      QPainter *painter, const QWidget *widget = Q_NULLPTR) const override;
```
Comment (MaciekPytel) [inline include/ui/veles_mainwindow.h]: I think we use nullptr (as opposed to Q_NULLPTR) in other places (including in this exact file, line 113). I think it's better to be consistent (I do assume Q_NULLPTR expands to nullptr anyway).

File: src/visualisation/panel.cc [hunk 2]
```diff
@@ -229,6 +230,9 @@ void VisualisationPanel::refreshVisualisation() {
 void VisualisationPanel::initLayout() {
   initOptionsPanel();
 
+  QWidget* root_widget = new QWidget(this);
+  this->setCentralWidget(root_widget);
```
Comment (MaciekPytel) [inline src/visualisation/panel.cc]: Why is this required?

File: include/ui/veles_mainwindow.h (hunk 1 above)
Comment (jacek-izykowski) [inline include/ui/veles_mainwindow.h]: Fixed.

File: src/visualisation/panel.cc (hunk 2 above)
Comment (jacek-izykowski) [inline src/visualisation/panel.cc]: That's because QMainWindow has a special built-in layout - QMainWindowLayout. QMainWindow::setCentralWidget sets main (central/"content") widget for this particular layout. QMainWindow is a base class for veles::ui::View because we plan to support dockable toolbars in the docks. And this particular feature is supported by QMainWindowLayout.

**Comment by MaciekPytel** [comment]
I'm not sure about putting the View class in veles_mainwindow. If nothing else I think it's strange to include veles_mainwindow in visualisation or hexedit. Intuitively I would more expect the include to be the other way round (with VelesMainWindow having method createHexEditTab). 
Maybe it would be better to put all dock-related stuff into a separate file and just keep main window itself in veles_mainwindow? Not sure.

**Comment by jacek-izykowski** [comment]
@MaciekPytel
veles_mainwindow.h/cc are now split into separate files (veles_mainwindow.h/cc and dockwidget.h/cc). VelesMainWindow is now accessed when it's actually needed.

File: CMakeLists.txt [hunk 3]
```diff
@@ -249,7 +250,8 @@ add_executable(main_ui
     ${SRC_DIR}/ui/main.cc
     ${SRC_DIR}/ui/logwidget.cc
     ${SRC_DIR}/ui/veles_mainwindow.cc
-    ${SRC_DIR}/ui/mainwindow_native.cc
+	${SRC_DIR}/ui/dockwidget.cc
```
Comment (adamiwaniuk) [inline CMakeLists.txt]: niespojne wciecie

File: CMakeLists.txt [hunk 4]
```diff
@@ -274,6 +276,7 @@ add_executable(main_ui
     ${CMAKE_SOURCE_DIR}/moc_commands.cpp
     ${CMAKE_SOURCE_DIR}/moc_digram.cpp
     ${CMAKE_SOURCE_DIR}/moc_veles_mainwindow.cpp
+	${CMAKE_SOURCE_DIR}/moc_dockwidget.cpp
```
Comment (adamiwaniuk) [inline CMakeLists.txt]: niespojne wciecie

File: include/visualisation/panel.h [hunk 5]
```diff
@@ -27,13 +27,14 @@
...
+#include "ui/dockwidget.h"
 #include "visualisation/base.h"
 #include "visualisation/minimap_panel.h"
 
 namespace veles {
 namespace visualisation {
 
-class VisualisationPanel : public QWidget {
+class VisualisationPanel : public veles::ui::View {
   Q_OBJECT
```
Comment (adamiwaniuk) [inline include/visualisation/panel.h]: do we need veles:: here or this kind of names are out convention?

Comment (jacek-izykowski) [inline include/visualisation/panel.h]: We don't. Fixed.

File: CMakeLists.txt (hunk 4 above)
Comment (jacek-izykowski) [inline CMakeLists.txt]: Fixed.

File: CMakeLists.txt (hunk 3 above)
Comment (jacek-izykowski) [inline CMakeLists.txt]: Fixed.
